File: 1859-sorting-the-sentence/1859-sorting-the-sentence.java
class Solution {
    public String sortSentence(String s) {
        String [] arr = s.split(" ");
        String [] newArray = new String [arr.length + 1];
        System.out.println(Arrays.toString(arr));
        for (int i = 0; i < arr.length; i++)
        {
        	int indx = Integer.parseInt(String.valueOf(arr[i].charAt(arr[i].length() - 1)));
        	String word = arr[i].substring(0, arr[i].length() - 1);
        	newArray[indx] = word;
        }
        String sorted = "";
        for (int i = 1; i < newArray.length; i++)
        {
        	sorted += newArray[i];
        	if (i < newArray.length - 1)
        		sorted += " ";
        }
        	
        return sorted;
    }
}

File: 1859-sorting-the-sentence/1859-sorting-the-sentence.py
class Solution:
    def sortSentence(self, s: str) -> str:
        arr = []
        for i in s:
            if i.isdigit():
                arr.append(0)
        word = ""
        for i in s:
            if i.isdigit():
                word.strip()
                arr[int(i) - 1] = word
                word = ""
            elif i == " ":
                continue
            else:
                word += i
        return " ".join(arr)